src/bluesearch/entrypoint/database/run.py

```diff
@@ -0,0 +1,636 @@
+                "SELECT article_id from articles WHERE article_id = :uid"
+            )
+            res = engine.execute(query, uid=uid).fetchall()
+
+            if not res:
+                new_uids.append(uid)
+
+        return not new_uids
+
+    def program_args(self) -> list[str]:
+        """Define subprocess arguments."""
+        input_dir = Path(self.input().path)
+
+        command = [
+            *BBS_BINARY,
+            "add",
+            *VERBOSITY,
+            self.db_url,
+            input_dir,
+            f"--db-type={self.db_type}",
+        ]
+
+        return command
+
+
+def get_all_dependencies(task: luigi.Task) -> set[luigi.Task]:
+    """Get all dependencies of a given task.
+
+    Parameters
+    ----------
+    task
+        Input task
+
+    Returns
+    -------
+    set[luigi.Task]
+        All the tasks that the `input` depends on including itself.
+    """
+    current_deps = set(task.deps())
+    if not current_deps:
+        return set()
+
+    else:
+        deps = {task}
+        for current_dep in current_deps:
+            deps |= get_all_dependencies(current_dep)
+
+        return deps | current_deps
+
+
+def run(
+    *,
+    source: str,
+    from_month: str,
+    filter_config: Path,
+    output_dir: Path,
+    db_url: str,
+    db_type: str,
+    mesh_topic_db: Path | None,
+    dry_run: bool,
+    grobid_host: str | None,
+    grobid_port: int | None,
+    identifier: str | None,
+    final_task: str | None,
+) -> int:
+    """Run overall pipeline.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    logger.info("Starting the overall pipeline")
+
+    DownloadTask.capture_output = CAPTURE_OUTPUT
+    UnzipTask.capture_output = CAPTURE_OUTPUT
+    TopicExtractTask.capture_output = CAPTURE_OUTPUT
+    TopicFilterTask.capture_output = CAPTURE_OUTPUT
+    PerformFilteringTask.capture_output = CAPTURE_OUTPUT
+    ConvertPDFTask.capture_output = CAPTURE_OUTPUT
+    ParseTask.capture_output = CAPTURE_OUTPUT
+    AddTask.capture_output = CAPTURE_OUTPUT
+
+    add_task_inst = AddTask(
+        source=source,
+        from_month=from_month,
+        filter_config=str(filter_config),
+        output_dir=str(output_dir),
+        mesh_topic_db=str(mesh_topic_db),
+        grobid_host=grobid_host,
+        grobid_port=grobid_port,
+        db_url=db_url,
+        db_type=db_type,
+        identifier=identifier,
+    )
+    if final_task is None:
+        selected_task_inst = add_task_inst
+    else:
+        all_dependencies = get_all_dependencies(add_task_inst)
+        all_dependencies_map = {t.__class__.__name__: t for t in all_dependencies}
+
+        if final_task in all_dependencies_map:
+            selected_task_inst = all_dependencies_map[final_task]
+        else:
+            raise ValueError(f"Unrecognized final task {final_task}")
+
+    luigi_kwargs = {
+        "tasks": [selected_task_inst],
+        "log_level": "WARNING",
```

Comment:
Would it make sense instead to look at the the verbosity level that is passed on the command line to the `bbs_database` command and apply it to `luigi` as well, instead of setting a default?

Reply:
Definitely something to be discussed. IMO there are 3 "namespaces" that we can control the verbosity of.


1) `luigi` itself
2) `bbs_database run` - basically all the `logger.info/logger.debug` inside of `entrypoint/database/run.py`
3) Subprocesses - basically all the `ExternalProgramTask` that are calling `bbs_database command`

The current status is that
1) Controlled via `bbs_database run -v`
2) Hardcoded to `WARNING`
3) Hardcoded to `-v` for all subprocesses

You are suggesting to merge 1) + 2) and IMO it is not good idea since `luigi` can be extremely verbose when the level is `INFO` or `DEBUG` and it might actually make it harder to "debug" problems in our code just by looking at the standard output.